lib/services/users.pm

```diff
@@ -113,7 +113,7 @@ sub switch_users {
     assert_screen "originUser-login-dm";
     #For poo#88247, we have to restore current user's password before migration,
     #so here need to use the original password.
-    type_password get_var('INCLUDE_SERVICES') ? "$password\n" : "$newpwd\n";
+    type_password(get_required_var('FLAVOR') =~ /Migration/ ? "$password\n" : "$newpwd\n");
```

Comment:
What's the different between 'get_var' and 'get_required_var'?

Reply:
See my reply to Richard. :)

Reply:
Thanks, given the fact that 'FlLAVOR' is a default variable there, 'get_var' can work file as well.
LGTM